src/cljs/main/broadfcui/common/gcs_file_preview.cljs

```diff
@@ -138,23 +139,35 @@
                                        {:error (.-responseText xhr)
                                         :status status-code}))
                     (when success?
-                      (let [content-type (:contentType (get-parsed-response))]
+                      (let [content-type (:contentType (get-parsed-response))
+                            scopes common/storage-scopes]
                         (when (previewable? object content-type)
-                          ;; TODO: get pet token from sam. Use pet token instead of (user/get-bearer-token-header) for
-                          ;; the following call.
-                          (ajax/call {:url (str "https://www.googleapis.com/storage/v1/b/" bucket-name "/o/"
-                                             (js/encodeURIComponent object) "?alt=media")
-                                      :headers (merge (user/get-bearer-token-header)
-                                                      {"Range" (str "bytes=-" preview-byte-count)})
-                                      :on-done (fn [{:keys [raw-response]}]
-                                                 (swap! state assoc :preview raw-response
-                                                        :preview-line-count (count (clojure.string/split raw-response #"\n+")))
-                                                 (after-update
-                                                  (fn []
-                                                    (when-not (string/blank? (@refs "preview"))
-                                                      (aset (@refs "preview") "scrollTop" (aget (@refs "preview") "scrollHeight"))))))})))))})))})
+                          (endpoints/call-ajax-sam
+                                      {:endpoint (endpoints/pet-token (:workspace-namespace props))
+                                       :payload scopes
+                                       :headers ajax/content-type=json
+                                       :on-done
+                                       (fn [{:keys [success? raw-response get-parsed-response status-code status-text]}]
+                                         (if success?
+                                           (let [pet-token (subs raw-response 1 (- (count raw-response) 1))]
```

Comment:
Is this simply removing the first character?  Why?

Reply:
I should have code-commented. Sam returns a quoted string, and we need the token without the quotes. See also https://github.com/broadinstitute/firecloud-orchestration/blob/develop/src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpSamDAO.scala#L44.